File: src/modules/models/SeatModel.js
class SeatModel {
    constructor() {
        this.seats = {};
    }

    initializeSeats(rows = ['A', 'B', 'C'], cols = 4) {
        rows.forEach(row => {
            for (let i = 1; i <= cols; i++) {
                const seatNumber = `${row}${i}`;
                this.seats[seatNumber] = {
                    seatNumber,
                    status: 'available',
                    lockedBy: null,
                    lockTimestamp: null
                };
            }
        });
    }

    getSeat(seatNumber) {
        return this.seats[seatNumber];
    }

    updateSeat(seatNumber, data) {
        Object.assign(this.seats[seatNumber], data);
    }

    getAllSeats() {
        return this.seats;
    }
}

module.exports = SeatModel;

File: src/modules/Booking&Seat Module/model/SeatModel.java

package model;

import java.util.HashMap;
import java.util.Map;

public class SeatModel {
    private Map<String, Seat> seats = new HashMap<>();

    public void initializeSeats() {
        char[] rows = {'A','B','C'};
        int cols = 4;

        for(char r: rows){
            for(int i=1;i<=cols;i++){
                String seat = r + String.valueOf(i);
                seats.put(seat, new Seat(seat));
            }
        }
    }

    public Seat getSeat(String seatNumber){
        return seats.get(seatNumber);
    }

    public Map<String, Seat> getAllSeats(){
        return seats;
    }
}
